app/routes/index.js

```diff
@@ -1,9 +1,39 @@
+require('dotenv').config();
 const express = require('express');
 const router = express.Router();
+const dotenv = require("dotenv");
+const jwt = require('jsonwebtoken');
+
+dotenv.config();
+
+process.env.TOKEN_SECRET;
 
 /* GET home page. */
 router.get('/', function(req, res, next) {
-    res.render('index', {title:'OG-YOGA'});
+    res.render('index', {title:'OG-YOGA', error: ""});
+});
+
+router.post('/login', async function(req, res, next) {
+    if(req.body.secret_key !== process.env.TOKEN_SECRET){
+        const error = "wrong secret key";
+        const token = await generateToken({username: req.body.username, error: error});
+        res.cookie('token', token, { httpOnly: true });
+        res.json({ token });
+    }else{
+        const token = await generateToken({username: req.body.username, error: ""});
+        res.cookie('token', token, { httpOnly: true });
+        res.json({ token });
+    }
 });
 
+//idk where to put this rn so I'll hold it here
+function generateToken(username, error) {
+    return jwt.sign({username: username, error: error}, process.env.TOKEN_SECRET, { expiresIn: '1800s'});
+}
+
+async function getTokenInfo(req, res, next) {
```

Comment:
The syntax of this function looks like a middleware function, if it is, move it to a folder called middlewares under app folder, and you can name the js file the same name as the function when exporting, just export the function itself, not as a json.

- not sure why you have to use headers right here, since token is being stored in cookie, you can just use req.cookies.token
- and check if the token is valid, if it is, call the next() function
- if it isn't you can redirect user to login page